### ISSUE TITLE
message delay

### ISSUE THREAD

**Issue by gplgpl775** [tdlib/td#3062]
Now I have a problem, I am creating a client using 'client = tdl.createClient(publicValue)' by 'var all_message = await client. invoke(
                        {
                            _: 'getChatHistory',
                            chat_id: itemQun.chat_id,
                            from_message_id: itemQun.from_message_id,
                            offset: -99, limit: 99
                            limit: 99
                        }
                    )' to get telegram messages (my program must use getChatHistory to get messages, because I am using vpn to access the extranet, and I want to realize that when the VPN is interrupted, and it returns to normal, I can continue to get the data according to the previous ids), in the initial stage of login, there is no delay in getting the messages, but when the After 3-4 days of running the program, some of the groups started to have delayed messages, so I decided to restart the project using node, and for about 1-2 two days these same groups had the same delay, so I deleted the login information and re-logged in, and persisted for 3-4 days, and then started to have the delayed messages again (it's always been a fixed number of groups). So what is the cause of this please?
It is important to note that there are many groups under the current account and the volume of group messages is also high.
Techniques: node; prebuilt-tdlib:“^0.1008034.0”; tdl:“^8.0.1”
Looking forward to your reply, thanks!

**Comment by gplgpl775** [comment]
It is important to note that there are many groups under the current account and the volume of group messages is also high

**Comment by levlam** [comment]
`getChatHistory` is not supposed to be used to fetch new messages.

Also, messages in supergroups and channels aren't supposed to be received immediately. They can be received by different users with different delay.

**Comment by gplgpl775** [comment]
I understand what you mean, but I have to use 'getChatHistory' again. So now I have a question. When I start a program, there is no delay at the beginning, but the delay of certain specific groups can reach 1 hour or even longer after 3-4 days. When I pause the program and restart, the delay disappears. What is the reason? @levlam

**Comment by levlam** [comment]
It can be anything and it is most likely to be intended. New messages are supposed to be received only through `updateNewMessage` and `getChatHistory` can't affect receiving of new messages.

**Comment by gplgpl775** [comment]
OK, I understand, thanks

> 它可以是任何东西，而且很可能是有意为之。新消息只能通过它接收`updateNewMessage`，`getChatHistory`不能影响新消息的接收。

**Comment by gplgpl775** [comment]
I am using "updateNewMessage" to get messages. The message volume is 150,000 per day. Why is there a problem of missing data? I opened two programs (the account has 150,000 daily messages), one using getChatHistory and the other using updateNewMessage. The difference in magnitude is 1/3. When I opened two programs again (the account has 30,000 daily messages), one using getChatHistory and the other using updateNewMessage, there is almost no difference in magnitude. What is the reason? What method should be used to get messages without delay and missing data?  @levlam
